jira/utils/__init__.py

```diff
@@ -79,3 +82,79 @@ def json_loads(resp: Response | None) -> Any:
         if not resp.text:
             return {}
         raise
+
+
+def cloud(client_method: Callable) -> Callable:
+    """A convenience decorator to check if the Jira instance is cloud.
+
+    Checks if the client instance is talking to Cloud Jira. If it is, return
+    the result of the called client method. If not, return None and log a
+    warning.
+
+    Args:
+      client_method: The method that is being called by the client.
+
+    Returns:
+      Either the result of the wrapped function or None.
+    """
+
+    @functools.wraps(client_method)
+    def check_if_cloud(*args, **kwargs):
+        # NOTE(jpavlav): The first argument of any class instance is a `self`
+        # reference. Avoiding magic numbers here.
+        instance = next(arg for arg in args)
+        if instance._is_cloud:
+            return client_method(*args, **kwargs)
+
+        instance.log.warning(
+            "This functionality is not available on Jira Data Center (Server) version."
+        )
+        return None
+
+    return check_if_cloud
+
+
+def experimental(client_method: Callable) -> Callable:
+    """A convenience decorator to inform if a client method is experimental.
```

Comment:
It might make sense to highlight experimental from which side ie ours or Atlassian here